web/src/components/shared/Ads/Board.tsx

```diff
@@ -28,23 +29,96 @@ export const URL_PARAMS: UrlParams<Omit<AdvertParams, 'lang'>> = {
   isOwnerVerificated: { name: 'verif', set: (v) => `${v}`, get: (v) => Boolean(v) },
 };
 
+export const adTypeToSeo: Record<AdvertType, SeoAdvertType> = {
+  purchase: 'buy',
+  selling: 'sell',
+};
+
+export const seoTypeToAd: Record<SeoAdvertType, AdvertType> = {
+  buy: 'purchase',
+  sell: 'selling',
+};
+
+const isSeoType = (type: string | undefined): type is SeoAdvertType => {
+  return Boolean(type && (type === 'buy' || type === 'sell'));
```

Comment:
```suggestion
  return type === 'buy' || type === 'sell';
```
Такой проверки достаточно